### PR TITLE
[WIP] alttree

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: include/veriblock/blockchain/pop/vbk_block_tree.hpp [hunk 1]
```diff
@@ -23,7 +23,17 @@ struct VbkBlockTree : public BlockTree<VbkBlock, VbkChainParams> {
...
+
+  VbkBlockTree(const VbkBlockTree& tree) = default;
+
+  VbkBlockTree& operator=(const VbkBlockTree& tree) {
+    this->activeChain_ = tree.activeChain_;
+    this->block_index_ = tree.block_index_;
+    this->fork_chains_ = tree.fork_chains_;
+    this->cmp_ = tree.cmp_;
+    return *this;
+  }
```
Comment (Warchant) [inline include/veriblock/blockchain/pop/vbk_block_tree.hpp:36]: remove this, and change `const &` of params to `const *`

File: src/blockchain/alt_block_tree.cpp [hunk 2]
```diff
@@ -90,19 +95,78 @@ bool AltTree::acceptBlock(const AltBlock& block,
   assert(index != nullptr &&
          "insertBlockHeader should have never returned nullptr");
 
+  if (!cmp_.proceedAllEndorsements(*index, context, state)) {
+    return state.Invalid("VbkTree::acceptBlock");
```
Comment (blackyblack) [inline src/blockchain/alt_block_tree.cpp:99]: Use BTC format for reject reason.

File: src/blockchain/pop/vbk_block_tree.cpp [hunk 3]
```diff
@@ -86,14 +86,7 @@ bool VbkBlockTree::acceptBlock(const VbkBlock& block,
...
-  // second, do stateless validation of payloads
-  for (const auto& p : payloads) {
-    if (!checkPayloads(p, state, getParams(), btc().getParams())) {
-      return state.Invalid("vbk-check-payloads");
-    }
-  }
-
-  if (!cmp_.proceedAllPayloads(*index, payloads, state)) {
+  if (!cmp_.proceedAllEndorsements(*index, context, state)) {
     return state.Invalid("VbkTree::acceptBlock");
```
Comment (blackyblack) [inline src/blockchain/pop/vbk_block_tree.cpp:90]: Same here.

File: src/blockchain/pop/vbk_block_tree.cpp [hunk 4]
```diff
@@ -86,15 +86,8 @@ bool VbkBlockTree::acceptBlock(const VbkBlock& block,
...
-  for (const auto& p : payloads) {
-    if (!checkPayloads(p, state, getParams(), btc().getParams())) {
-      return state.Invalid("vbk-check-payloads");
-    }
-  }
-
-  if (!cmp_.proceedAllPayloads(*index, payloads, state)) {
-    return state.Invalid("VbkTree::acceptBlock");
+  if (!cmp_.proceedAllEndorsements(*index, context, state)) {
+    return state.Invalid("vbk-procedd-all-endorsements-block");
```
Comment (blackyblack) [inline src/blockchain/pop/vbk_block_tree.cpp:90]: proceed

Comment (Warchant) [reply src/blockchain/pop/vbk_block_tree.cpp:90]: @blackyblack it will be fixed in next PR

**Pull request merged**